medmodels/data_synthesis/synthesizer_model.py

```diff
@@ -0,0 +1,84 @@
+"""This module defines the SynthesizerModel class, which is a PyTorch module that
+is used to generate synthetic data using a synthesizer model, after the synthesizer
+model has been trained on real data."""
+
+from abc import ABCMeta, abstractmethod
+from pathlib import Path
+from typing import Any, Dict, Union
+
+import sparse
+import torch
+from torch import nn
+
+from medmodels import MedRecord
+from medmodels.data_synthesis.synthesizer import Synthesizer
+
+
+class SynthesizerModel(nn.Module, metaclass=ABCMeta):
+    """SynthesizerModel is an abstract class that serves as a blueprint for synthesizer models.
+
+    This class is instantiated once a Synthesizer has been trained on real data. It is
+    separated from the Synthesizer class to allow for the generation of synthetic data
+    without incurring into side effects."""
```

Comment:
also sounds very chatgpt.

`allow for the generation of synthetic data without incurring into side effects.`
this sentence does not make sense and there still are side effects, just more expected ones